File: lib/ts/src/internal/validation/ValidateString.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { getTypeUnexpectedValidationFailure } from '../../internal/validation/GetTypeUnexpectedValidationFailure.js';
import { ValidationFailure } from '../../internal/validation/ValidationFailure.js';
import { stringName } from '../types/TString.js';

export function validateString(value: any): ValidationFailure[] {
    if (typeof value === 'string') {
        return [];
    } else {
        return getTypeUnexpectedValidationFailure([], value, stringName);
    }
}

File: lib/py/telepact/internal/validation/ValidateString.py
#|
#|  Copyright The Telepact Authors
#|  SPDX-License-Identifier: Apache-2.0
#|

from ..types.TString import _STRING_NAME
from ...internal.validation.ValidationFailure import ValidationFailure
from ...internal.validation.GetTypeUnexpectedValidationFailure import get_type_unexpected_validation_failure


def validate_string(value: object) -> list['ValidationFailure']:
    if type(value) is str:
        return []
    if isinstance(value, str):
        return []
    return get_type_unexpected_validation_failure([], value, _STRING_NAME)
